lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
+  return positiveMod(val, 360);
+}
+
+/**
+ * Normalize discrepancies between Sass color spaces and ColorJS color space
+ * ids.
+ */
+function encodeSpaceForColorJs(space?: KnownColorSpace): string | undefined {
+  switch (space) {
+    case 'rgb':
+      return 'srgb';
+    case 'a98-rgb':
+      return 'a98rgb';
+    case 'display-p3':
+      return 'p3';
+    case 'prophoto-rgb':
+      return 'prophoto';
+  }
+  return space;
+}
+
+/**
+ * Normalize discrepancies between Sass color spaces and ColorJS color space
+ * ids.
+ */
+function decodeSpaceFromColorJs(space: string, isRgb = false): KnownColorSpace {
+  switch (space) {
+    case 'srgb':
+      return isRgb ? 'rgb' : space;
+    case 'xyz-d65':
+      return 'xyz';
+    case 'a98rgb':
+      return 'a98-rgb';
+    case 'p3':
+      return 'display-p3';
+    case 'prophoto':
+      return 'prophoto-rgb';
+  }
+  return space as KnownColorSpace;
+}
+
+// @TODO For some spaces (e.g. Lab and Oklab), ColorJS only accepts `l` and not
+// `lightness` as a channel name. Maybe a bug?
+/**
+ * Normalize discrepancies between Sass channel names and ColorJS channel ids.
+ */
+function encodeChannelForColorJs(channel: ChannelName): string {
+  if (channel === 'lightness') return 'l';
+  return channel;
+}
+
+/**
+ * Implement our own check of channel name validity for a given space, because
+ * ColorJS allows e.g. `b` for either `blue` or `blackness` or `b` channels.
+ */
+function validateChannelInSpace(
+  channel: ChannelName,
+  space: KnownColorSpace
+): void {
+  if (channel === 'alpha') return;
+  let valid = false;
+  switch (space) {
+    case 'rgb':
+    case 'srgb':
+    case 'srgb-linear':
+    case 'display-p3':
+    case 'a98-rgb':
+    case 'prophoto-rgb':
+    case 'rec2020':
+      valid = ['red', 'green', 'blue'].includes(channel);
+      break;
+    case 'hsl':
+      valid = ['hue', 'saturation', 'lightness'].includes(channel);
+      break;
+    case 'hwb':
+      valid = ['hue', 'whiteness', 'blackness'].includes(channel);
+      break;
+    case 'lab':
+    case 'oklab':
+      valid = ['lightness', 'a', 'b'].includes(channel);
+      break;
+    case 'lch':
+    case 'oklch':
+      valid = ['lightness', 'chroma', 'hue'].includes(channel);
+      break;
+    case 'xyz':
+    case 'xyz-d65':
+    case 'xyz-d50':
+      valid = ['x', 'y', 'z'].includes(channel);
+      break;
+  }
+  if (!valid) {
+    throw valueError(
+      `Unknown channel name "${channel}" for color space "${space}".`
+    );
+  }
+}
+
+/** Determine whether the given space is a polar color space. */
+function isPolarColorSpace(space: KnownColorSpace): space is PolarColorSpace {
+  switch (space) {
+    case 'hsl':
+    case 'hwb':
+    case 'lch':
+    case 'oklch':
+      return true;
+    default:
+      return false;
+  }
+}
+
+/**
+ * Normalize between ColorJS coordinates (which use `NaN`) and Sass Color
+ * coordinates (which use `null`).
+ */
+function getCoordsFromColor(
```

Comment:
Can you clarify that this is specifically going from the colorjs representation to the Sass representation? The name and documentation are currently kind of ambiguous. (Passing in a colorjs color would also work.)

Reply:
Renamed and added comments in https://github.com/sass/embedded-host-node/pull/259/commits/18ab10ee84ce3b3321b8a8b98f52c7fb79650653